sequencer/src/l1_client.rs

```diff
@@ -164,6 +163,7 @@ impl L1Client {
     /// and `new`. Returns `Vec<FeeInfo>`
     pub async fn get_finalized_deposits(
         &self,
+        fee_contract_address: Address,
```

Comment:
Could we pass this to the L1Client on construction instead or is that inconvenient?

Reply:
I was trying to isolate state into the `ChainConfig` struct. Eventually, the validation logic will need to keep track of two different chain configs and switch between them when we reach an upgrade. This makes it easy to just pass the new fee address into the L1 client, rather than having to mutate or create a new L1 client.